syntax = "proto3";

package optuna.study;

import "pruners.proto";
import "samplers.proto";

service Study {
  rpc create(CreateStudyOptions) returns (StudyInstance) {}
}

message CreateStudyOptions {
  string study_name = 1;
  optuna.samplers.Sampler sampler = 2;
  optuna.pruners.Pruner pruner = 3;
  string storage = 4;
  Direction direction = 5;
}

message StudyInstance {
  uint64 instance_id = 1;
}

enum  Direction {
  MINIMIZE = 0;
  MAXIMIZE = 1;
}